services/datamanager/builtin/capture_test.go

```diff
@@ -0,0 +1,166 @@
+		},
+		{
+			name:            "Capture should work for remotes too.",
+			remoteCollector: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// Set up capture directories.
+			initCaptureDir, err := os.MkdirTemp("", "")
+			test.That(t, err, test.ShouldBeNil)
+			defer func() {
+				err := os.RemoveAll(initCaptureDir)
+				test.That(t, err, test.ShouldBeNil)
+			}()
+			updatedCaptureDir, err := os.MkdirTemp("", "")
+			test.That(t, err, test.ShouldBeNil)
+			defer func() {
+				err := os.RemoveAll(updatedCaptureDir)
+				test.That(t, err, test.ShouldBeNil)
+			}()
+
+			// Set up robot config.
+			var initConfig *config.Config
+			if tc.remoteCollector {
+				initConfig = setupConfig(t, remoteCollectorConfigPath)
+			} else if tc.initialCollectorDisableStatus {
+				initConfig = setupConfig(t, disabledTabularCollectorConfigPath)
+			} else {
+				initConfig = setupConfig(t, enabledTabularCollectorConfigPath)
+			}
+
+			// Set up service config.
+			initSvcConfig, ok1, err := getServiceConfig(initConfig)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, ok1, test.ShouldBeTrue)
+			initSvcConfig.CaptureDisabled = tc.initialServiceDisableStatus
+			initSvcConfig.ScheduledSyncDisabled = true
+			initSvcConfig.CaptureDir = initCaptureDir
+
+			// Build and start data manager.
+			dmsvc := newTestDataManager(t)
+			err = dmsvc.Update(context.Background(), initConfig)
+			time.Sleep(captureTime)
+
+			// Set up updated robot config.
+			var updatedConfig *config.Config
+			if tc.newCollectorDisableStatus {
+				updatedConfig = setupConfig(t, disabledTabularCollectorConfigPath)
+			} else {
+				updatedConfig = setupConfig(t, enabledTabularCollectorConfigPath)
+			}
+
+			// Set up updated service config.
+			updatedServiceConfig, ok, err := getServiceConfig(updatedConfig)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, ok, test.ShouldBeTrue)
+			updatedServiceConfig.CaptureDisabled = tc.newServiceDisableStatus
+			updatedServiceConfig.ScheduledSyncDisabled = true
+			updatedServiceConfig.CaptureDir = updatedCaptureDir
+
+			// Update to new config and let it run for a bit.
+			err = dmsvc.Update(context.Background(), updatedConfig)
+			test.That(t, err, test.ShouldBeNil)
+			time.Sleep(captureTime)
+
+			// Check if initial config captured (or not) as we'd expect.
+			initialCaptureFiles := getAllFiles(initCaptureDir)
+			if !tc.initialServiceDisableStatus && !tc.initialCollectorDisableStatus {
+				// TODO: check contents
+				test.That(t, len(initialCaptureFiles), test.ShouldBeGreaterThan, 0)
+			} else {
+				test.That(t, len(initialCaptureFiles), test.ShouldEqual, 0)
+			}
+
+			// Check if updated config captured (or not) as we'd expect.
+			updatedCaptureFiles := getAllFiles(updatedCaptureDir)
+			if !tc.newServiceDisableStatus && !tc.newCollectorDisableStatus {
+				//TODO: check contents
+				test.That(t, len(updatedCaptureFiles), test.ShouldBeGreaterThan, 0)
```

Comment:
I believe there was previously a check that the content size > empty sensor reading proto size. If that's not easy to add, then can keep in the TODO.

Reply:
Done